README.md

```diff
@@ -1,4 +1,196 @@
+3. Implement [`JibMavenPluginExtension`](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin-extension-api/src/main/java/com/google/cloud/tools/jib/maven/extension/JibMavenPluginExtension.java) (Maven) / [`JibGradlePluginExtension`](https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin-extension-api/src/main/java/com/google/cloud/tools/jib/gradle/extension/JibGradlePluginExtension.java) (Gradle).
+
+### Updating Container Build Plan
+
+The extension API passes in [`ContainerBuildPlan`](https://github.com/GoogleContainerTools/jib/blob/master/jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/ContainerBuildPlan.java), which is the container build plan originally prepared by Jib plugins. The build plan describes in a declarative way how it plans to build a container image.
+
+The class is a Java API for [Container Build Plan Specification](https://github.com/GoogleContainerTools/jib/blob/master/proposals/container-build-plan-spec.md). The Container Build Plan Specification is a general specification independent of Jib. Likewise, the Container Build Plan Java API is a light-weight, standalone API implementing the spec published to Maven Central ([`jib-build-plan`](https://search.maven.org/artifact/com.google.cloud.tools/jib-build-plan). The Build Plan classes, once instantiated, are all stateless, immutable "value classes" (holding only simple values). You can inspect the values using simple getters, and when you want to "modify" values, use `toBuilder()` to create new instances.
+
+### Defining Extension-Specific Configuration
+
+Sometimes, you may want to make your extension configurable by the extension end-users. See #using-jib-plugin-extensions to understand how end-users can provide extra configuration to an extension.
+
+- Simple string properties (map): the Extension API has a built-in support for end-users passing simple string map. If your extension does not need complex configuration structure, prefer this approach.
+
+- Complex configuration structure: define your configuration class and have `getExtraConfigType()` return the class. See the [Maven](first-party/jib-ownership-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/ownership/Configuration.java) and [Gradle](https://github.com/GoogleContainerTools/jib-extensions/blob/master/first-party/jib-ownership-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/ownership/Configuration.java) examples.
+   - Gradle-specific: your configuration class must have a 1-arg constructor accepting a Gradle [`Project`](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html).
+
+### Version Matrix
+
+| jib-maven-plugin | jib-maven-plugin-extension-api |
+|:----------------:|:------------------------------:|
+| 2.3.0 - current  | 0.3.0                          |
```

Comment:
should this just be 2.4.0 for uniformity?

Reply:
I've mulled over this. I think this is OK, even if we didn't announce it before. And it won't matter much anyway when we make more releases like 2.5.0, 2.6.0, etc. This may help for those who cannot upgrade from 2.3.0 to 2.4.0 for some reason (although I admit this is a silly excuse). If you still think we should go for 2.4.0, feel free to file a PR. This doc needs to be updated a few times anyway.